daemon/cmd/daemon_main.go

```diff
@@ -1682,13 +1767,14 @@ func initKubeProxyReplacementOptions() {
 		}
 	}
 
-	if option.Config.EnableNodePort &&
-		option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled {
-		if option.Config.Tunnel != option.TunnelDisabled {
-			log.Fatalf("Cannot use NodePort acceleration with tunneling. Either run cilium-agent with --%s=%s or --%s=%s",
-				option.NodePortAcceleration, option.NodePortAccelerationDisabled, option.TunnelName, option.TunnelDisabled)
-		}
+	if !option.Config.EnableNodePort {
+		// Make sure that NodePort dependencies are disabled
+		disableNodePort()
+	}
 
+	// After this point, BPF NodePort should not be disabled
```

Comment:
Then, can we return early if `!option.Config.EnableNodePort` and avoid the two `if` blocks conditioned on `option.Config.EnableNodePort` below?